src/components/CollectionsAndPalettes/common/ItemCard/ItemCard.tsx

```diff
@@ -2,19 +2,41 @@ import * as React from "react";
 import cn from "classnames";
 import styles from "../../common/ItemCard/ItemCard.module.scss";
 import { CollectionItemT } from "../../../../constants/types";
-import { AppContext, AppContextState } from "../../../../context/AppContext";
+import {
+  AppContext,
+  AppContextState,
+  Modals,
+} from "../../../../context/AppContext";
 import { get } from "lodash-es";
+import { matchPath, RouteComponentProps, withRouter } from "react-router-dom";
+import { COLLECTION_URL } from "../../../../constants/urls";
+import { deleteItem } from "../../../../context/CollectionsAPI/api";
 
-interface Props {
+interface ItemProps {
   mode: any;
   item: CollectionItemT;
 }
 
-export default class ItemCard extends React.Component<Props, any> {
+type Props = RouteComponentProps;
+
+class ItemCard extends React.Component<Props & ItemProps, any> {
   static contextType = AppContext;
   context!: AppContextState;
   materialItem: any;
 
+  getCollectionId = () => {
+    const collectionPageResult = matchPath(this.props.location.pathname, {
+      path: COLLECTION_URL,
+      exact: true,
+    });
+    return get(collectionPageResult, "params.collection_id");
+  };
+
+  deleteItem = () => {
+    this.context.storeItemId(this.props.item.id);
+    this.context.openModal(Modals.DeleteItem);
```

Comment:
You can use the modalProps here instead of storing the id manually.

```
this.context.openModal(Modals.DeleteItem, {collectionItemId: this.props.item.id})
```
And then, in the modal, access it with
```
const modalParams = this.context.getModalParams()
modalParams.collectionItemId <-- should have the id
```